resources/js/Pages/Collections/Components/CollectionsHeading/RefreshButton.tsx

```diff
@@ -0,0 +1,74 @@
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { IconButton } from "@/Components/Buttons";
+import { Tooltip } from "@/Components/Tooltip";
+import { useAuthorizedAction } from "@/Hooks/useAuthorizedAction";
+import { useToasts } from "@/Hooks/useToasts";
+import { isTruthy } from "@/Utils/is-truthy";
+
+export const RefreshButton = ({ wallet }: { wallet: App.Data.Wallet.WalletData | null }): JSX.Element => {
+    const [loading, setLoading] = useState(false);
+    const [disabled, setDisabled] = useState(false);
+    const { t } = useTranslation();
+
+    const { signedAction } = useAuthorizedAction();
+    const { showToast } = useToasts();
+
+    const refresh = (): void => {
+        void signedAction(async ({ authenticated }) => {
+            if (authenticated) {
```

Comment:
The `authenticated` parameter (or, in some cases, you can also pass the `signed` parameter) represents the state before signing in or logging in. One use case for this parameter is when a user attempts to like a gallery while not logged in. In such cases, we don't know whether the user had previously liked the gallery, so this parameter is used to force a like action instead of toggling the value.

In this specific case, I believe we don't need to check for the authenticated parameter (you can't even access this section if you are not authenticated). Therefore, you can remove that condition. In other words, if the method runs, it implies the user is authenticated AND signed.

Side note: Perhaps we should consider renaming these parameters to usedToBeAuthenticated and usedToBeSigned to avoid confusion. What do you think?